content/livesync/connector/database.textile

```diff
@@ -0,0 +1,198 @@
+| @channel@ | String | Yes | @documents@ | The "channel:/channels name on which to publish this change event. |
+| @name@ | String | Yes | @edit@ | The message event name to use when publishing the message. |
+| @rejected@ | Boolean | No (defaults to FALSE) | @FALSE@ | True if the event rejects a client side change, false to confirm the change. Defaults to false (confirming the change). |
+| @data@ | JSON | Yes | @{ "read": false, "data": { "timestamp": 1674744488658, "body": "Lorem ipsum" } }@ | The message payload to use when publishing the message. |
+| @headers@ | JSON | No (optional) | @{ "id": 123, "type": "document", "author": "socrates", "pages": [1, 5, 7] }@ | A set of message attributes, provided under the headers key in the message extras. These are optional properties that may contain metadata and/or ancillary payloads. |
+| @locked_by@ | String | No (implementation detail) | @0d6c0277-e88a-4dba-a854-e80a4bd75317@ | The ID of the node that has locked this record. This is an implementation detail of how the Database Connector processes records and you should not set a value for this column when inserting an outbox record. |
+| @lock_expiry@ | Timestamp | No (implementation detail) | @2023-06-12 16:24:27 @| The timestamp at which the lock will expire. This is an implementation detail of how the Database Connector processes records and you should not set a value for this column when inserting an outbox record. |
+
+h3(#change-detection). Change detection
+
+The Database Connector uses a poll-on-change strategy to query for new records to process when they are inserted into the outbox. This is the strategy used with the @ADBC_POLL_FIXED_RATE@ configuration option set to @false@. This is achieved through the use of a "trigger":https://www.postgresql.org/docs/current/sql-createtrigger.html configured on the outbox table and is the default behavior. The trigger invokes a "function":https://www.postgresql.org/docs/current/sql-createfunction.html#:~:text=Use%20CREATE%20OR%20REPLACE%20FUNCTION%20to%20change%20a%20function%20definition,the%20owner%20of%20the%20function. which uses "NOTIFY":https://www.postgresql.org/docs/current/sql-notify.html to broadcast a notification to the Database Connector, which it receives using "LISTEN":https://www.postgresql.org/docs/current/sql-listen.html.
+
+Internally, the Database Connector debounces notifications within a window determined by the @ADBC_POLL_INTERVAL@ "configuration option":/livesync/connector#options.
+
+This approach avoids imposing additional load on the database due to polling when there are no new records to process.
+
+Alternatively, you can configure the Database Connector to periodically poll for new records with an interval determined by the @ADBC_POLL_INTERVAL@ configuration option by setting @ADBC_POLL_FIXED_RATE@ to @true@.
```

Comment:
Could we add a line here that says:

> In the future we intend to support change detection via [logical decoding](https://www.postgresql.org/docs/current/logicaldecoding-explanation.html). If you're interested in this functionality, please reach out <some link>.

Reply:
Ideally we should only be documenting functionality as it exists today. If we're planning on adding a feature in the future and want some input into it, then I'd suggest it's added as a question in the feedback form rather than throughout the usage instructions. Happy to pick this one up one outside of this PR.